features/step_definitions/adding_event_instances.rb

```diff
@@ -0,0 +1,36 @@
+Given(/^I am on the 'Add an Event' page$/) do
+  visit new_event_instance_path
+end
+
+Given(/^there is not an event called 'NuConf'$/) do
+  if ev = Event.find_by(name:'NuConf')
+    ev.destroy
+  end
+end
+
+Given(/^there is an event called (\w+) that does not have an instance for the year (\d+)$/) do |conf, year|
+  event = create(:event, name: conf)
+  if inst = event.instances.find_by(year: year)
```

Comment:
Hmm...I find this technique, interesting. Never seen it before.

I'm torn by it — as I think when running the test, there shouldn't be anything left in the DB...and I'd want to know if things were being left over...

And yet — if they were, we wouldn't find out via this test..

What about if we changed them to expectations, like: expect(Event.find_by(name: etc..).to eq(false)?

Is that strange because we're running expectations before the end of the test script?

Let me know what you think.

Reply:
So... I had started responding to all of these, and in a previous life of this comment, had talked myself in a loop into thinking this was unnecessary.

TL;DR is that we should trust `database_cleaner` to do its job, and also assume that Factory Bot won't instantiate a related model apropos of nothing, unless we tell it to.

I think I did this because I wanted the code to reflect the step definition: there *is* an event, and it *must not* have an instance for xxxx year. But I think it's too literal now on reflection. Will change.

Reply:
Okay, cool. :-)